com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProtocolServer.java

```diff
@@ -38,9 +41,8 @@
      */
     public ProtocolServer(InputStream input, OutputStream output, IProviderContext context) {
         super(input, output);
-        this.debugAdapter = new DebugAdapter((message) -> {
-            sendMessage(message);
-        }, context);
+        IDebugAdapterContext debugContext = new DebugAdapterContext(this, context);
```

Comment:
The DebugAdapter should be responsible for providing the DebugAdapterContext. The adapter wants the handlers to have some knowledge about their runtime environment, which is the adapter itself.